steam_wrapper/steam_wrapper.py

```diff
@@ -170,8 +171,11 @@ def migrate_data():
         copytree(source, target, ignore=[steam_root_source])
         if os.path.isdir(steam_root_source):
             os.rename(steam_root_source, steam_root_target)
```

Comment:
Won't this still fail with overrides? I actually wonder if we could do migrate, then tell flatpak to restart the app at the end.

Reply:
It will fail on `xdg-data/Steam` override, yes.

Reply:
Actually I wonder why we can rename the parent directory of the mount point, but not the mount point itself. Anyway, I still consider passing `~/.local/share/Steam` from host a bad idea (it is not shareable with non-flatpak steam).

Reply:
Then IMHO we need to check and block. The failure is quite confusing. Also I don't like the idea of *copying* mounted things. I think I did it wrong before as well. I think ideally

1. Copy things where not a mount
2. Move directory aside
3. Create symlink
4. execv flatpak-spawn the same script so we get a new sandbox
5. Remove the directory that was moved aside
6. Continue

Reply:
Err, this PR wasn't about handling the steam root dir, it was for safely handling cases like `xdg-data/icons`.
I agree that we need to exclude mounted dirs, it's just out of scope here.

Reply:
Ok.

Reply:
I'm exposing the restart in https://github.com/flathub/com.valvesoftware.Steam/pull/702; can you rebase this after I merge that and hook into the restart mechanism?